### PR TITLE
Removes confusing zero mask from VilBERT

### PR DESCRIPTION
**Pull request by dirkgr**
I found this while trolling through @jacob-morrison's code. That mask is unused later. If it were used, it would be wrong.

### CODE REVIEW

**Comment by jacob-morrison** [comment]
I think we'll have problems in the bimodal encoder if we don't pass in the mask: https://github.com/allenai/allennlp/blob/main/allennlp/modules/transformer/bimodal_encoder.py#L194-L199. I think the problem is actually in the encoder though (ie we should still delete this useless mask in the backbone), because `co_attention_mask`'s default value is `None`, but I'm pretty sure these lines will break if it is `None`.

**Review by jacob-morrison — CHANGES_REQUESTED** [review]
I think we'll have problems in the bimodal encoder if we don't pass in the mask: https://github.com/allenai/allennlp/blob/main/allennlp/modules/transformer/bimodal_encoder.py#L194-L199. I think the problem is actually in the encoder though (ie we should still delete this useless mask in the backbone), because `co_attention_mask`'s default value is `None`, but I'm pretty sure these lines will break if it is `None`.

**Comment by dirkgr** [comment]
I think that fixes it, but let's see what the tests say.

**Review by jacob-morrison — APPROVED** [review]
LGTM!
